### PR TITLE
[WIP] Automate configuration changes to optimize API-M profiles

### PR DESCRIPTION
**Pull request by sarubi**
## Purpose
The PR contains changes to automate the profile setup configurations while starting an APIM profile. Up to now, it has been manually done by users when they wanted to run a selected profile.

## Documentation
[https://docs.google.com/document/d/1nB112HRslaqXLQA50lXdHblf42QgMTIzTCAPVaOFjrA/edit?usp=sharing](url)

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: modules/distribution/product/src/main/startup-scripts/profileSetup.sh [hunk 1]
```diff
@@ -0,0 +1,296 @@
+#!/bin/bash
```
Comment (pubudu538) [inline modules/distribution/product/src/main/startup-scripts/profileSetup.sh]: Let's add the license header!

**Comment by prasa7** [comment]
Shall we add the PR description mentioning how to run and so on.

File: modules/distribution/product/src/main/startup-scripts/profileSetup.sh [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+		disableJMSConnectionDetails
+		disableTransportSenderWS
+		disableTransportSenderWSS
+		removeWebSocketInboundEndpoint
+		removeSecureWebSocketInboundEndpoint
+		removeSynapseConfigs
+		#remove unnecessary webapps
+		if [ "$(find $pathToWebapps -type f -not -name 'api#am#publisher#v0.12.war' | wc -l)" -gt 0 ]
+		then
+			find $pathToWebapps -type f -not -name 'api#am#publisher#v0.12.war' -delete
```
Comment (Krishanx92) [inline modules/distribution/product/src/main/startup-scripts/profileSetup.sh]: APIM 2.5.0 it should be v0.13.war. better to use regex for this. check other places as well

File: modules/distribution/product/src/main/startup-scripts/profileSetup.sh (hunk 1 above)
Comment (sarubi) [reply modules/distribution/product/src/main/startup-scripts/profileSetup.sh]: Resolved

File: modules/distribution/product/src/main/startup-scripts/profileSetup.sh (hunk 2 above)
Comment (sarubi) [reply modules/distribution/product/src/main/startup-scripts/profileSetup.sh]: Resolved

**Pull request merged**